vault/resource_mfa_duo.go

```diff
@@ -0,0 +1,149 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+func mfaDuoResource() *schema.Resource {
+	return &schema.Resource{
+		Create: mfaDuoWrite,
+		Update: mfaDuoWrite,
+		Delete: mfaDuoDelete,
+		Read:   mfaDuoRead,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Name of the MFA method.",
+				ValidateFunc: validateNoTrailingSlash,
+			},
+			"mount_accessor": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The mount to tie this method to for use in automatic mappings. The mapping will use the Name field of Aliases associated with this mount as the username in the mapping.",
+			},
+			"username_format": {
```

Comment:
It would be great to get test coverage of this field, as well as `push_info`.

Reply:
Done! 👍